packages/documentation/copy/pt/reference/Decorators.md

```diff
@@ -0,0 +1,573 @@
+> Veja [Metadados](#metadados) para mais informações sobre a biblioteca `reflet-metadata`.
+
+## Decoradores de Parâmetros
+
+Um _Decorador de Parâmetros_ é declarado antes de uma declaração de parâmetro.
+O decorador de parâmetro é aplicado à função para um construtor de classe ou declaração de método.
+Um decorador de parâmetro não pode ser usado em um arquivo de declaração, uma sobrecarga ou em qualquer outro contexto de ambiente (como em uma classe `declare`).
+
+A expressão para o decorador de parâmetro será chamada como uma função em tempo de execução, com os três argumentos a seguir:
+
+1. A função construtora da classe para um membro estático ou o protótipo da classe para um membro de instância.
+2. O nome do membro.
+3. O índice ordinal do parâmetro na lista de parâmetros da função.
+
+> NOTA&emsp; Um decorador de parâmetro só pode ser usado para observar que um parâmetro que foi declarado em um método.
+
+O valor de retorno do decorador de parâmetro é ignorado.
+
+A seguir está um exemplo de um decorador de parâmetro (`@requerido`) aplicado ao parâmetro de um membro da classe` Recepcionista`:
+
+```ts
+class Recepcionista {
+  cumprimento: string;
+
+  constructor(mensagem: string) {
+    this.cumprimento = mensagem;
+  }
+
+  @validar
+  cumprimentar(@requerido nome: string) {
+    return "Olá " + nome + ", " + this.cumprimento;
+  }
+}
+```
+
+Podemos então definir os decoradores `@requerido` e` @validar` usando as seguintes declarações de função:
+
+```ts
+import "reflect-metadata";
+
+const chaveDeMetodosNecessaria = Symbol("requerido");
+
+function requerido(
+  alvo: Object,
+  chaveDePropriedade: string | symbol,
+  indiceDeParametro: number
+) {
+  let parametrosNecessariosExistentes: number[] =
+    Reflect.getOwnMetadata(chaveDeMetodosNecessaria, alvo, chaveDePropriedade) || [];
+  parametrosNecessariosExistentes.push(indiceDeParametro);
+  Reflect.defineMetadata(
+    chaveDeMetodosNecessaria,
+    parametrosNecessariosExistentes,
+    alvo,
+    chaveDePropriedade
+  );
+}
+
+function validar(
+  alvo: any,
+  nomeDaPropriedade: string,
+  descritor: DescritorDePropriedadeTipada<Function>
+) {
+  let método = descritor.value;
+  descritor.value = function () {
+    let parametrosObrigatorios: number[] = Reflect.getOwnMetadata(
+      chaveDeMetodosNecessaria,
+      alvo,
+      nomeDaPropriedade
+    );
+    if (parametrosObrigatorios) {
+      for (let indiceDeParametro of parametrosObrigatorios) {
+        if (
+          indiceDeParametro >= arguments.length ||
+          arguments[indiceDeParametro] === undefined
+        ) {
+          throw new Error("Argumento obrigatório ausente.");
+        }
+      }
+    }
+
+    return method.apply(this, arguments);
+  };
+}
+```
+
+O decorador `@requerido` adiciona uma entrada de metadados que marca o parâmetro como necessário.
+O decorador `@validar` então envolve o método` cumprimentar` existente em uma função que valida os argumentos antes de invocar o método original.
```

Comment:
```suggestion
A seguir está um exemplo de um decorador de parâmetro (`@obrigatorio`) aplicado ao parâmetro de um membro da classe` Recepcionista`:

```ts
class Recepcionista {
  cumprimento: string;

  constructor(mensagem: string) {
    this.cumprimento = mensagem;
  }

  @validar
  cumprimentar(@obrigatorio nome: string) {
    return "Olá " + nome + ", " + this.cumprimento;
  }
}
```

Podemos então definir os decoradores `@obrigatorio` e` @validar` usando as seguintes declarações de função:

```ts
import "reflect-metadata";

const chaveDeMetodosNecessaria = Symbol("obrigatorio");

function obrigatorio(
  alvo: Object,
  chaveDePropriedade: string | symbol,
  indiceDeParametro: number
) {
  let parametrosNecessariosExistentes: number[] =
    Reflect.getOwnMetadata(chaveDeMetodosNecessaria, alvo, chaveDePropriedade) || [];
  parametrosNecessariosExistentes.push(indiceDeParametro);
  Reflect.defineMetadata(
    chaveDeMetodosNecessaria,
    parametrosNecessariosExistentes,
    alvo,
    chaveDePropriedade
  );
}

function validar(
  alvo: any,
  nomeDaPropriedade: string,
  descritor: DescritorDePropriedadeTipada<Function>
) {
  let método = descritor.value;
  descritor.value = function () {
    let parametrosObrigatorios: number[] = Reflect.getOwnMetadata(
      chaveDeMetodosNecessaria,
      alvo,
      nomeDaPropriedade
    );
    if (parametrosObrigatorios) {
      for (let indiceDeParametro of parametrosObrigatorios) {
        if (
          indiceDeParametro >= arguments.length ||
          arguments[indiceDeParametro] === undefined
        ) {
          throw new Error("Argumento obrigatório ausente.");
        }
      }
    }

    return method.apply(this, arguments);
  };
}
```

O decorador `@obrigatorio` adiciona uma entrada de metadados que marca o parâmetro como necessário.
O decorador `@validar` então envolve o método `cumprimentar` existente em uma função que valida os argumentos antes de invocar o método original.
```

I think `obrigatorio` is a better option than `requerido`

Reply:
Truth, mandatory is really a better option than required, I will update according to the changes and resend the PR.
I appreciate the criticisms suggested 😉👍